### PR TITLE
Doc fixes for attributes and customer_profile API sections

### PR DESCRIPTION
**Pull request by anuraj-optimizely**
@darwishoptly @ajithopti @sramanan @mdodsworth 
Fixes for Attributes and Customer Profiles API section

### CODE REVIEW

File: src/content/rest/customer_profiles/customer_profiles/0-intro.md [hunk 1]
```diff
@@ -4,15 +4,15 @@ title: Customer Profiles
 anchor: customer_profiles
 ---
 
-Customer Profiles are a collection of customer (your users) attributes across Datasources.
+Customer Profiles are a collection of customer user's attributes across Datasources.
```
Comment (sramanan) [inline src/content/rest/customer_profiles/customer_profiles/0-intro.md]: users' - plural possessive

File: src/content/rest/customer_profiles/customer_profiles/0-intro.md [hunk 2]
```diff
@@ -4,15 +4,15 @@ title: Customer Profiles
...
 
-The following APIs are designed to create, update and read customer attributes for a single datasource at a time.
+The following APIs are designed to create, update and read customer's attributes data for a single datasource at a time.
```
Comment (sramanan) [inline src/content/rest/customer_profiles/customer_profiles/0-intro.md]: customers' - plural possessive

File: src/content/rest/customer_profiles/customer_profiles/0-intro.md (hunk 1 above)
Comment (ajithopti) [inline src/content/rest/customer_profiles/customer_profiles/0-intro.md]: This is hard to parse. Because developer docs are written keeping our customers as the reader, you can write: `Customer Profiles are a collection of your customers' attributes across several Datasources. The following APIs are designed to create, update and read customer attributes for a single datasource at a time.`

File: src/content/rest/customer_profiles/dcpattributes/1-read.md [hunk 3]
```diff
@@ -7,24 +7,25 @@ type: GET
...
-  archived: Whether Attribute has been archived
+  dcp_datasource_id: ID of parent datasource
+  archived: Boolean flag whether an Attribute has been archived
   description: Description of the attribute
-  format: When datatype is date, format is date format
-  datatype: Datatype of Attribute. Can be "string", "bool", "long", "double", "datetime"
-  version: API version
+  format: When datatype is date, format can be "yyyy-mm-dd", "yyyy-mm-ddThh:mm:ssZ", "epoch"
+  datatype: Datatype of the Attribute. Can be "string", "bool", "long", "double", "datetime"
+  version: Version of the attribute, currently attribute incremental versioning is not supported
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/1-read.md]: Break run-on sentence into 2: `attribute, currently` -> `attribute. Currently`.

File: src/content/rest/customer_profiles/dcpattributes/1-read.md [hunk 4]
```diff
@@ -7,24 +7,25 @@ type: GET
...
-  dcp_datasource_id: Id of parent datasource
-  archived: Whether Attribute has been archived
+  dcp_datasource_id: ID of parent datasource
+  archived: Boolean flag whether an Attribute has been archived
   description: Description of the attribute
-  format: When datatype is date, format is date format
-  datatype: Datatype of Attribute. Can be "string", "bool", "long", "double", "datetime"
-  version: API version
+  format: When datatype is date, format can be "yyyy-mm-dd", "yyyy-mm-ddThh:mm:ssZ", "epoch"
+  datatype: Datatype of the Attribute. Can be "string", "bool", "long", "double", "datetime"
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/1-read.md]: Use emphatic language when specifying choices. `Can be "string", ...` -> `Must be one of "string", ...`

File: src/content/rest/customer_profiles/dcpattributes/1-read.md [hunk 5]
```diff
@@ -7,24 +7,25 @@ type: GET
...
+  version: Version of the attribute, currently attribute incremental versioning is not supported
+  name: Name of the Attribute, use this name to upload customer_profile data using REST API or bulk upload. Note that attribute name is case sensitive
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/1-read.md]: Break run-on sentence into 2: `Attribute, use` -> `Attribute. Use`

Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/1-read.md]: More precise: `Use this name when you upload...`

File: src/content/rest/customer_profiles/dcpattributes/2-create.md [hunk 6]
```diff
@@ -19,20 +19,20 @@ response: |
...
+- "datetime": needs format to be a Datetime format
 
-#### Datetime formats
-- ISO_8601_DATE = 'yyyy-mm-dd'
-- ISO_8601_DATE_TIME_UTC = 'yyyy-mm-ddThh:mm:ssZ'
-- EPOCH = 'epoch'
+#### Datetime format options
+- "yyyy-mm-dd": ISO_8601 UTC date format
+- "yyyy-mm-ddThh:mm:ssZ": ISO_8601 datetime format
+- "epoch": Epoch time in millis
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/2-create.md]: `millis` -> `milliseconds`

File: src/content/rest/customer_profiles/customer_profiles/0-intro.md (hunk 1 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/customer_profiles/0-intro.md]: Yeah, I prefer that. I'd also perhaps drop the 'designed' and `at a time`.

`The following APIs allow you to create, update and read customer attributes for a single datasource`

File: src/content/rest/customer_profiles/customer_profiles/0-intro.md [hunk 7]
```diff
@@ -4,15 +4,15 @@ title: Customer Profiles
...
 
-To use these APIs make sure you have read the sections on [DCPServices](/rest/customer_profiles#customer_profile_service), [Datasources](/rest/customer_profiles#customer_profile_service), and [Attributes](/rest/customer_profiles#dcp_attributes)
+To use these APIs make sure you have read the sections on [DCPServices](/rest/customer_profiles#customer_profile_service), [Datasources](/rest/customer_profiles#dcp_datasources
```
Comment (mdodsworth) [inline src/content/rest/customer_profiles/customer_profiles/0-intro.md]: `To use these APIs, it's recommended that you first read the sections on`

File: src/content/rest/customer_profiles/customer_profiles/0-intro.md (hunk 1 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/customer_profiles/0-intro.md]: I know this is how it is everywhere but why did we go with `Datasource`, rather than `data source`?

File: src/content/rest/customer_profiles/dcpattributes/3-update.md [hunk 8]
```diff
@@ -23,9 +23,12 @@ response: |
...
-Updates Attribute Metadata for a given Datasource
+Updates metadata for an Attribute. 
 
 #### Editable fields
-- `datatype`
 - `description`
-- `format`
+
+<div class="lego-attention lego-attention--warning push--bottom">
+*NOTE:* Updates to attribute's name or datatype or version is not supported currently. If you want to edit name or datatype or format archive the attribute and create a new one.
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/3-update.md]: You may specify the conjunction `or` before the last choice. So: `Updates to an attribute's name, datatype, or version are not supported currently; if you wish to edit these properties, archive the attribute and create a new one with the desired properties`

File: src/content/rest/customer_profiles/dcpattributes/4-delete.md [hunk 9]
```diff
@@ -7,6 +7,6 @@ endpoint_option: 789
 title: Delete Attribute
 anchor: delete-dcpdatasource
 ---
-Delete a Attribute
+Archives the Attribute
 
-Once an Attribute is deleted, you can no longer send or read Customer Profile data for this Attribute 
+Once an Attribute is archived, you can no longer send or read Customer Profile data for this Attribute. You can unarchive Attribute at any time.
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/4-delete.md]: `You can unarchive Attribute at any time.` -> `You can unarchive an Attribute at any time.`

File: src/content/rest/customer_profiles/customer_profiles/1-read.md [hunk 10]
```diff
@@ -8,11 +8,14 @@ anchor: read-customer_profile
...
+    },
+    "dcpServiceId": "567",
+    "datasourceId": "678",
+    "customerId": "oeu1234.5678"
   }
 ---
 
 Retrieves customer profile attributes for given dcp_service_id, dcp_datasource_id and customer_id.
-A customer_id is a Datasources' identifier for a given customer
+A customer_id is a Datasource's identifier for a given customer
```
Comment (mdodsworth) [inline src/content/rest/customer_profiles/customer_profiles/1-read.md]: `customer_id represents the Datasource specific identifier for a customer`?

File: src/content/rest/customer_profiles/dcpattributes/5-list_attributes.md [hunk 11]
```diff
@@ -6,31 +6,26 @@ endpoint_option: 678
...
+      "format": null,
+      "datatype": "string",
+      "version": 1,
+      "id": 790,
+      "name": "most_viewed_category"
+    }
   ]
 ---
-Fetches all Attributes and Metadata for a given Datasource
+Fetches all attributes for a given Datasource. [GET Datasource](/rest/customer_profiles/index.html#read-dcpdatasource) also returns all attributes for a given datasource.
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/5-list_attributes.md]: Might as well use `Get` or `Gets` consistently instead of `Fetches`.

File: src/content/rest/customer_profiles/customer_profiles/0-intro.md [hunk 12]
```diff
@@ -4,15 +4,15 @@ title: Customer Profiles
...
+<img src="/assets/img/dcp/customer_profiles.png">), and [Attributes](/rest/customer_profiles#dcp_attributes)
 
-<img src="/assets/img/dcp/customer_profiles.png">
 
-Using [/consolidated_customer_profile_view](/rest/customer_profiles#consolidated-profile), you can retrieve the full Customer Profile across Datasources.
+Using [/consolidated_customer_profile_view](/rest/customer_profiles#consolidated-profile), you can retrieve the full Customer's Profile across Datasources.
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/customer_profiles/0-intro.md]: `, you can retrieve the complete Customer Profile consolidated across all Datasources.`

File: src/content/rest/customer_profiles/customer_profiles/2-write.md [hunk 13]
```diff
@@ -1,31 +1,33 @@
...
+    "datasourceId": "678",
+    "customerId": "oeu1234.5678"
   }
 ---
 
-Creates / Updates attributes for customer profiles, given dcp_service_id, dcp_datasource_id and customer_id.
+Creates/Updates attributes for a customer profile given `dcp_service_id`, `dcp_datasource_id` and `customer_id`.
 
-Takes a set of key value pairs, where each key corresponds to the `name` of the [Attribute](/rest/customer_profiles#dcp_attributes).
+Takes a set of key value pairs, where each key corresponds to the `name` of the [Attribute](/rest/customer_profiles#dcp_attributes) and `value` is provided in attribute's original datatype and format.
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/customer_profiles/2-write.md]: `and the "value" conforms to the datatype and format specified for the attribute.`

File: src/content/rest/customer_profiles/customer_profiles/2-write.md [hunk 14]
```diff
@@ -1,31 +1,33 @@
...
 
 #### Note:
 - If an Attribute already has a value for a given customer_id, the value for the given Attribute is updated
 - A subset of defined [Attribute](/rest/customer_profiles#dcp_attributes) is ok.
 - If a column header does not correspond to a [Attribute](/rest/customer_profiles#dcp_attributes) `name`, the write will fail
-- If an Attribute value does respect the [Attribute](/rest/customer_profiles#dcp_attributes) `datatype`, the write will fail
+- If an Attribute value does respect the [Attribute](/rest/customer_profiles#dcp_attributes) `datatype`/`format`, the write will fail
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/customer_profiles/2-write.md]: `If an Attribute value does respect the...` -> `If an Attribute value does not respect the...`

File: src/content/rest/customer_profiles/customer_profiles/2-write.md [hunk 15]
```diff
@@ -1,31 +1,33 @@
...
-Creates / Updates attributes for customer profiles, given dcp_service_id, dcp_datasource_id and customer_id.
+Creates/Updates attributes for a customer profile given `dcp_service_id`, `dcp_datasource_id` and `customer_id`.
 
-Takes a set of key value pairs, where each key corresponds to the `name` of the [Attribute](/rest/customer_profiles#dcp_attributes).
+Takes a set of key value pairs, where each key corresponds to the `name` of the [Attribute](/rest/customer_profiles#dcp_attributes) and `value` is provided in attribute's original datatype and format.
 
 #### Note:
 - If an Attribute already has a value for a given customer_id, the value for the given Attribute is updated
 - A subset of defined [Attribute](/rest/customer_profiles#dcp_attributes) is ok.
 - If a column header does not correspond to a [Attribute](/rest/customer_profiles#dcp_attributes) `name`, the write will fail
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/customer_profiles/2-write.md]: `correspond to a ` -> `correspond to an `

File: src/content/rest/customer_profiles/customer_profiles/2-write.md [hunk 16]
```diff
@@ -1,31 +1,33 @@
...
   }
 ---
 
-Creates / Updates attributes for customer profiles, given dcp_service_id, dcp_datasource_id and customer_id.
+Creates/Updates attributes for a customer profile given `dcp_service_id`, `dcp_datasource_id` and `customer_id`.
 
-Takes a set of key value pairs, where each key corresponds to the `name` of the [Attribute](/rest/customer_profiles#dcp_attributes).
+Takes a set of key value pairs, where each key corresponds to the `name` of the [Attribute](/rest/customer_profiles#dcp_attributes) and `value` is provided in attribute's original datatype and format.
 
 #### Note:
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/customer_profiles/2-write.md]: Be consistent; end each sentence with a period.

File: src/content/rest/customer_profiles/dcpattributes/0-intro.md [hunk 17]
```diff
@@ -4,9 +4,9 @@ title: Attributes
 anchor: dcp_attributes
 ---
 
-Attributes are customer descriptors for a given Datasource. The Attribute APIs below are used to register Attributes and manage metadata for a given Datasource.
+The Attributes APIs below are used to register Attributes and manage attribute's metadata for a given Datasource.
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/0-intro.md]: `to register Attributes and manage attribute's metadata for a given Datasource.` -> `to register and manage Attribute's metadata for a given Datasource.`

File: src/content/rest/customer_profiles/customer_profiles/2-write.md (hunk 13 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/customer_profiles/2-write.md]: I think `is provided in attribute's original datatype and format` might be confusing. We may want to be more specific about the conversion in a separate section that we reference.

I'd also switch this around to be:

`Takes a set of key value pairs containing the Attribute name and associated value. The `value` can be provided in attribute's original datatype and format. `

File: src/content/rest/customer_profiles/dcpattributes/1-read.md [hunk 18]
```diff
@@ -7,24 +7,25 @@ type: GET
 title: Read Attribute
 anchor: read-dcpattribute
 fields:
-  dcp_datasource_id: Id of parent datasource
-  archived: Whether Attribute has been archived
+  dcp_datasource_id: ID of parent datasource
+  archived: Boolean flag whether an Attribute has been archived
```
Comment (ajithopti) [inline src/content/rest/customer_profiles/dcpattributes/1-read.md]: `Boolean flag whether an ` -> `Boolean indicating whether an `

File: src/content/rest/customer_profiles/customer_profiles/2-write.md (hunk 15 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/customer_profiles/2-write.md]: not you: `an Attribute`

File: src/content/rest/customer_profiles/customer_profiles/2-write.md (hunk 14 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/customer_profiles/2-write.md]: `does not respect`

File: src/content/rest/customer_profiles/dcpattributes/0-intro.md (hunk 17 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/0-intro.md]: I'd go with `attribute metadata`

File: src/content/rest/customer_profiles/dcpattributes/0-intro.md [hunk 19]
```diff
@@ -4,9 +4,9 @@ title: Attributes
...
 
 <img src="/assets/img/dcp/Attributes.png">
 In the example above, the Attributes are all highlighted in yellow.
 
-[Customer Profile](/rest/customer_profiles/#customer_profiles) APIs are used to upload and update attribute values for specific customers.
+[Customer Profile](/rest/customer_profiles/#customer_profiles) APIs are used to upload and update customer profile attributes data.
```
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/0-intro.md]: attribute

File: src/content/rest/customer_profiles/dcpattributes/1-read.md (hunk 4 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/1-read.md]: I'd be tempted to cover data types and the supported input formats (incl. conversion) in a separate section that we reference

File: src/content/rest/customer_profiles/dcpattributes/1-read.md [hunk 20]
```diff
@@ -7,24 +7,25 @@ type: GET
...
     "last_modified": "2015-08-18T21:38:55.927680Z",
     "version": 1,
-    "id": 3324671622,
-    "name": "is_high_value_customer"
+    "id": 789,
+    "name": "LTV"
   }
 ---
-Reads Attribute Metadata for a given DCP Datasource
+Get metadata for a single attribute.
```
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/1-read.md]: get the

File: src/content/rest/customer_profiles/dcpattributes/1-read.md (hunk 5 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/1-read.md]: run-on sentence into 2: `Name of the Attribute. Used to identify an attribute across our REST APIs and bulk upload`

File: src/content/rest/customer_profiles/dcpattributes/2-create.md [hunk 21]
```diff
@@ -19,20 +19,20 @@ response: |
...
-Creates Attribute Metadata for a given Datasource
+Creates Attribute for a given Datasource
 
 #### Datatype options
 - "string"
 - "bool"
 - "long"
 - "double"
-- "datetime"
+- "datetime": needs format to be a Datetime format
```
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/2-create.md]: ah, you did move the datetime type information into its own section. Excellent. Perhaps `a supported Datime format` with link to below section.

File: src/content/rest/customer_profiles/dcpattributes/3-update.md [hunk 22]
```diff
@@ -23,9 +23,12 @@ response: |
     "name": "admin_account_id"
   }
 ---
-Updates Attribute Metadata for a given Datasource
+Updates metadata for an Attribute.
```
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/3-update.md]: the metadata

File: src/content/rest/customer_profiles/dcpattributes/3-update.md (hunk 8 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/3-update.md]: `Updates to an attribute's name, datatype, or version are not currently supported. We recommend archiving the existing attribute and creating a replacement, with the desired name, datatype, and format.`

I'm splitting an infinitive but I think it reads better.

File: src/content/rest/customer_profiles/dcpattributes/4-delete.md (hunk 9 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/4-delete.md]: update, rather than send?

File: src/content/rest/customer_profiles/dcpattributes/5-list_attributes.md (hunk 11 above)
Comment (mdodsworth) [inline src/content/rest/customer_profiles/dcpattributes/5-list_attributes.md]: nit: we seem to be capitalizing `Datasource` elsewhere.

**Comment by tscanlin** [comment]
@anuraj-optimizely can this be merged?

**Comment by tscanlin** [comment]
I'm going to assume this PR can be abandoned since it hasn't seen any action in a long time. If it shouldn't just respond and say it should stay open, otherwise I will close it and leave the branch on github at the end of the week.
